tests/test_gap.py

```diff
@@ -667,3 +667,29 @@ def test_data_shorter_than_length() -> None:
             {},
             None,
         )
+
+
+def test_data_shorter_than_required_length():
+    """Test data that is too short."""
+
+    data = (b"\x02 a\xc4|\x04@*'\x9c\xa5C\r\xa1\xe6\x1e\xe7\x8f\xa57D\xe6$\x03",)
```

Comment:
Maybe parametrize the test to test more of the error paths in `_uncached_parse_advertisement_data`, or does this somehow trigger all the error paths?
Also:
- Consider adding a comment for the data to mention how it's invalid
- I find the representation as mixed ascii and escaped hex to be really hard to read, but maybe that doesn't matter

Reply:
I just pushed a parametrize with all the bad data I discovered